### PR TITLE
Fix #2351

### LINKED ISSUE #2351: OBDScanner Unit Tests need refactoring
All OBDScanner unit tests weren't executed for a long time through the silent change to XENIAL runners on travis. 

This is a reminder to refactor the following unit tests:
https://github.com/secdev/scapy/blob/master/test/tools/obdscanner.uts
https://github.com/secdev/scapy/blob/master/test/contrib/automotive/obd/scanner.uts

### PR DESCRIPTION
**Pull request by TabeaSpahn**
fixes #2351

### CODE REVIEW

File: test/tools/obdscanner.uts [hunk 1]
```diff
@@ -225,8 +225,7 @@ responder.start()
 
 
 = Test DTC scan
-eprint("TODO: this test is currently failing. It has to be reworked")
-exit(0)
+
```
Comment (guedou) [inline test/tools/obdscanner.uts]: This test is failing due to this change. Could you revert your modification?

File: test/tools/obdscanner.uts [hunk 2]
```diff
@@ -104,14 +102,16 @@ if p.wait() == 0:
 + Syntax check
 
 = Import isotp
-conf.contribs['ISOTP'] = {'use-can-isotp-kernel-module': False}
+conf.contribs['ISOTP'] = {'use-can-isotp-kernel-module': ISOTP_KERNEL_MODULE_AVAILABLE}
```
Comment (guedou) [inline test/tools/obdscanner.uts]: Could you explain theses changes?

File: test/tools/obdscanner.uts [hunk 3]
```diff
@@ -259,30 +253,26 @@ s1_pid03 = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID03(fuel_sys
 s1_pid0B = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0B(data=100)])
 s1_pid0F = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0F(data=50)])
 s1_queue.put(s1_pid00)
-s1_queue.put(s1_pid0F)
```
Comment (guedou) [inline test/tools/obdscanner.uts]: Why are you changing the order and the next results?

File: test/tools/obdscanner.uts [hunk 4]
```diff
@@ -259,30 +253,26 @@ s1_pid03 = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID03(fuel_sys
 s1_pid0B = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0B(data=100)])
 s1_pid0F = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0F(data=50)])
 s1_queue.put(s1_pid00)
```
Comment (polybassa) [inline test/tools/obdscanner.uts]: In my opinion, the design of this tests is not perfect. There are better and more reliable ways to get identical test cases

File: test/tools/obdscanner.uts (hunk 1 above)
Comment (TabeaSpahn) [reply test/tools/obdscanner.uts]: I am still trying to fix that test.
I did another commit that should fix this test without ignoring the test.

File: test/tools/obdscanner.uts (hunk 2 above)
Comment (TabeaSpahn) [reply test/tools/obdscanner.uts]: I did not see a reason why this should be `False`. There is a variable that is `True`  if the kernel module exists. And in my opinion if it exists it should be used.

File: test/tools/obdscanner.uts (hunk 3 above)
Comment (TabeaSpahn) [reply test/tools/obdscanner.uts]: I tried fixing the tests, apparently the order of the messages does matter for this test.
I now comited a more robust version of this, were the order is not relevant anymore.

File: test/tools/obdscanner.uts (hunk 4 above)
Comment (TabeaSpahn) [reply test/tools/obdscanner.uts]: See new commit.

File: test/contrib/automotive/obd/scanner.uts [hunk 5]
```diff
@@ -140,111 +135,111 @@ from subprocess import call
...
-with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
-    ready.wait(timeout=5)
-    all_ids_set = set(range(1, 256))
-    supported_ids = _supported_id_numbers(socket, 0.1, OBD_S01, 'pid', False)
-    unsupported_ids = all_ids_set - supported_ids
++ Simulate scanner
 
 = Run scanner
+
 exit_if_no_isotp_module()
```
Comment (polybassa) [inline test/contrib/automotive/obd/scanner.uts]: You should be able to remove this call. The initial setup should work on every Travis system, and on Windows.

**Comment by polybassa** [comment]
Thanks for your refactoring. The unit tests look much cleaner, now.

File: test/contrib/automotive/obd/scanner.uts [hunk 6]
```diff
@@ -140,111 +135,111 @@ from subprocess import call
...
+with ISOTPSocket("vcan0", 0x7e8, 0x7e0, basecls=OBD, padding=True) as ecu:
+    answering_machine = ECU_am(supported_responses=example_responses, main_socket=ecu, basecls=OBD)
+    sim = threading.Thread(target=answering_machine, kwargs={'count': 12})
+    sim.start()
+    try:
+        with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
+            all_ids_set = set(range(1, 256))
+            supported_ids = _supported_id_numbers(socket, 0.1, OBD_S01, 'pid', False)
+            unsupported_ids = all_ids_set - supported_ids
+
```
Comment (polybassa) [inline test/contrib/automotive/obd/scanner.uts]: I had problems with empty lines on indented code blocks in UTScapy, in the past. Just as remark.

**Comment by guedou** [comment]
@polybassa what is you call on this PR?

**Comment by polybassa** [comment]
I like the refactoring. I’m fine with this PR. Thanks Tabea

**Comment by polybassa** [comment]
This PR looks good from my side

**Pull request merged**